### PR TITLE
Post updates should expect slashed data

### PR DESCRIPTION
**Pull request by joehoyle**
Right now, updating posts with quotes causes slashes to be saved to the db, as
wp_update_post does `wp_slash` on already slashed data if you pass it an
object.

I'm not sure if this is the best fix, but the unit test illustrates the issue.

### CODE REVIEW

File: tests/test-rest-posts-controller.php [hunk 1]
```diff
@@ -712,6 +712,18 @@ public function test_create_post_with_invalid_date_gmt() {
...
+	public function test_create_post_with_quotes_in_title() {
+		wp_set_current_user( $this->editor_id );
+
+		$request = new WP_REST_Request( 'POST', '/wp/v2/posts' );
+		$params = $this->set_post_data( array(
+			'title' => "Rob O'Rourke's Diary",
+		) );
+		$request->set_body_params( $params );
+		$response = $this->server->dispatch( $request );
+		$this->check_create_post_response( $response );
```
Comment (danielbachhuber) [inline tests/test-rest-posts-controller.php]: Can we have an explicit test for the response title, to make this test more precise?

Comment (danielbachhuber) [inline tests/test-rest-posts-controller.php]: 301f33b

Comment (sanchothefat) [inline tests/test-rest-posts-controller.php]: I approve of this test. Also worth testing emoji and other unicode characters.
